src/Features/CSharp/Portable/ConvertPrimaryToRegularConstructor/ConvertPrimaryToRegularConstructorCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,504 @@
+
+                    foreach (var identifierName in grouping)
+                    {
+                        var xmlElement = identifierName.AncestorsAndSelf().OfType<XmlEmptyElementSyntax>().FirstOrDefault();
+                        if (xmlElement is { Name.LocalName.ValueText: "paramref" })
+                        {
+                            var seeTag = xmlElement
+                                .ReplaceToken(xmlElement.Name.LocalName, Identifier("see").WithTriviaFrom(xmlElement.Name.LocalName))
+                                .WithAttributes(SingletonList<XmlAttributeSyntax>(XmlCrefAttribute(
+                                    TypeCref(fieldName))));
+
+                            editor.ReplaceNode(xmlElement, seeTag);
+                        }
+                        else
+                        {
+                            editor.ReplaceNode(identifierName, fieldName.WithTriviaFrom(identifierName));
+                        }
+                    }
+                }
+            }
+        }
+
+        void FixParameterAndBaseArgumentListIndentation()
+        {
+            var currentRoot = mainDocumentEditor.GetChangedRoot();
+            var formattingOptions = optionsProvider.GetOptions(document.Project.Services).CleanupOptions.FormattingOptions;
+            var indentationOptions = new IndentationOptions(formattingOptions);
+
+            var formattedRoot = Formatter.Format(currentRoot, SyntaxAnnotation.ElasticAnnotation, solution.Services, formattingOptions, cancellationToken);
+
+            var constructor = (ConstructorDeclarationSyntax)formattedRoot.GetAnnotatedNodes(constructorAnnotation).Single();
+
+            var rewrittenParameterList = AddElementIndentation(typeDeclaration, constructor, constructor.ParameterList, static list => list.Parameters);
+            var initializer = constructor.Initializer;
+            var rewrittenInitializer = initializer?.WithArgumentList(AddElementIndentation(typeDeclaration, constructor, initializer.ArgumentList, static list => list.Arguments));
+
+            var rewrittenConstructor = constructor
+                .WithParameterList(rewrittenParameterList)
+                .WithInitializer(rewrittenInitializer);
+
+            var rewrittenRoot = formattedRoot.ReplaceNode(constructor, rewrittenConstructor);
+            mainDocumentEditor.ReplaceNode(mainDocumentEditor.OriginalRoot, rewrittenRoot);
+        }
+
+        static TListSyntax AddElementIndentation<TListSyntax>(
+            TypeDeclarationSyntax typeDeclaration,
+            ConstructorDeclarationSyntax constructorDeclaration,
+            TListSyntax list,
+            Func<TListSyntax, IEnumerable<SyntaxNode>> getElements)
+            where TListSyntax : SyntaxNode
+        {
+            // Since we're moving parameters from the constructor to the type, attempt to dedent them if appropriate.
+
+            var typeLeadingWhitespace = GetLeadingWhitespace(typeDeclaration);
```

Comment:
Will formatter do the job here?

Reply:
No. And I've found trying to depend on the formatter for precise manipulation of existing code is often not viable.  All the latest code fixers that work with indentation (use file scoped namespace, use collection expression, use primary constructor, etc.) all avoid it.